pkg/rest/coverage/swagger_analysis_test.go

```diff
@@ -0,0 +1,175 @@
+				Expect(restAPIStats).To(Equal(expectedRestAPI))
+			})
+
+		})
+
+		Context("With URI filter", func() {
+
+			It("Should build filtered REST API structure", func() {
+				expectedRestAPI := map[string]map[string]*RequestStats{
+					"/pets/{name}": map[string]*RequestStats{
+						"GET": &RequestStats{
+							Body:         nil,
+							Query:        map[string]int{},
+							ParamsHit:    0,
+							ParamsNum:    0,
+							MethodCalled: false,
+							Path:         "/pets/{name}",
+							Method:       "GET",
+						},
+						"DELETE": &RequestStats{
+							Body:         nil,
+							Query:        map[string]int{},
+							ParamsHit:    0,
+							ParamsNum:    0,
+							MethodCalled: false,
+							Path:         "/pets/{name}",
+							Method:       "DELETE",
+						},
+						"PATCH": &RequestStats{
+							Body:         map[string]int{},
+							Query:        map[string]int{},
+							ParamsHit:    0,
+							ParamsNum:    6,
+							MethodCalled: false,
+							Path:         "/pets/{name}",
+							Method:       "PATCH",
+						},
+					},
+				}
+
+				restAPIStats, err := getRESTApiStats(petStoreSwaggerPath, "/pets/{name}")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(restAPIStats).To(Equal(expectedRestAPI))
+
+			})
+		})
+
+		It("Should add swagger params", func() {
+			document, err := loads.JSONSpec(petStoreSwaggerPath)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Resolving referenced definitions")
+			reqStats := RequestStats{
+				Body:         nil,
+				Query:        map[string]int{},
+				ParamsHit:    0,
+				ParamsNum:    0,
+				MethodCalled: false,
+				Path:         "/pets",
+				Method:       "POST",
+			}
+			expectedReqStats := reqStats
+			expectedReqStats.Body = map[string]int{}
+			expectedReqStats.ParamsNum = 6
+
+			addSwaggerParams(&reqStats, document.Analyzer.ParamsFor("POST", "/pets"), document.Spec().Definitions)
+			Expect(reqStats).To(Equal(expectedReqStats))
+
+			By("Not resolving referenced definitions")
+			reqStats = RequestStats{
+				Body:         nil,
+				Query:        map[string]int{},
+				ParamsHit:    0,
+				ParamsNum:    0,
+				MethodCalled: false,
+				Path:         "/pets",
+				Method:       "GET",
+			}
+			expectedReqStats = reqStats
+			expectedReqStats.Query = map[string]int{
+				"tags":  0,
+				"limit": 0,
+			}
+			expectedReqStats.ParamsNum = 2
+
+			addSwaggerParams(&reqStats, document.Analyzer.ParamsFor("GET", "/pets"), document.Spec().Definitions)
+			Expect(reqStats).To(Equal(expectedReqStats))
+		})
+
+		It("Should count params from referenced models", func() {
+			document, err := loads.JSONSpec(petStoreSwaggerPath)
+			Expect(err).NotTo(HaveOccurred())
+
+			params := document.Analyzer.ParamsFor("POST", "/pets")
+			pCnt := countRefParams(params["body#Pet"].Schema, document.Spec().Definitions)
+			Expect(pCnt).To(Equal(6))
+
+			pCnt = countRefParams(params["body#Pet"].Schema, spec.Definitions{})
+			Expect(pCnt).To(Equal(0))
```

Comment:
Would you please add descriptors for value based Expectes? Without them the expects are not that informative. Thank you.